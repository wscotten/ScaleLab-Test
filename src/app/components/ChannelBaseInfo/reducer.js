import { GET_CHANNEL_INFO } from './constants';

const initialData = {
  name: '',
  status: 'linked',
  youtube_channel_id: '',
  total_views: 0,
  subscribers: 0,
  videos: 0,
  last_month_views: 0,
  network: '',
  joined_at: '',
  owner: '',
  category: '',
  commission: 0,
  country_id: '',
};

const channelBaseInfoReducer = (data = initialData, { type, payload = {} }) => {
  switch (type) {
    case GET_CHANNEL_INFO:
      return payload.channelBaseInfo;
    default:
      return data;
  }
};

export default channelBaseInfoReducer;
