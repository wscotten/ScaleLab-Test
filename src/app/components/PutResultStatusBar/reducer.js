import {
  PUT_SOCIAL_REACH_INFO_FAIL,
  PUT_SOCIAL_REACH_INFO_SUCCESS,
} from '../../containers/SocialReachModalButton/reducer';
import { CLOSE_SOCIAL_REACH_MODAL } from '../../containers/SocialReachEditButton/constants';

const putResultStatusReducer = (putStatusResult = null, { type }) => {
  switch (type) {
    case CLOSE_SOCIAL_REACH_MODAL:
      return null;
    case PUT_SOCIAL_REACH_INFO_SUCCESS:
      return true;
    case PUT_SOCIAL_REACH_INFO_FAIL:
      return false;
    default:
      return putStatusResult;
  }
};

export default putResultStatusReducer;
