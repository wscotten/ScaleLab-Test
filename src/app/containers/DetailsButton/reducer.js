export const TOGGLE_DETAILS_BUTTON = 'TOGGLE_DETAILS_BUTTON';

export const toggleDetailsButton = () => ({
  type: TOGGLE_DETAILS_BUTTON,
});

const detailsButtonReducer = (currentDetailsState = true, { type }) => {
  switch (type) {
    case TOGGLE_DETAILS_BUTTON:
      return !currentDetailsState;
    default:
      return currentDetailsState;
  }
};

export default detailsButtonReducer;
