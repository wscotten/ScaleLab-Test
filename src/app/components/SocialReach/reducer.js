import { PUT_SOCIAL_REACH_INFO_SUCCESS } from '../../containers/SocialReachModalButton/reducer';

export const GET_CHANNEL_INFO = 'GET_CHANNEL_INFO';

export const extractSocialReachNumbers = payload => ({
  total: payload.channelStats.social_reach,
  Youtube: payload.channelStats.youtube_reach,
  Facebook: payload.channelStats.facebook_reach,
  Vine: payload.channelStats.vine_reach,
  Twitter: payload.channelStats.twitter_reach,
  Instagram: payload.channelStats.instagram_reach,
});

const ageGroupsReducer = (socialReach = {}, { type, payload }) => {
  switch (type) {
    case GET_CHANNEL_INFO:
      return extractSocialReachNumbers(payload);
    case PUT_SOCIAL_REACH_INFO_SUCCESS:
      return payload;
    default:
      return socialReach;
  }
};

export default ageGroupsReducer;
