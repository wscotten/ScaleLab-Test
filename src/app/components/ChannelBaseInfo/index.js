import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import moment from 'moment';
import ProfilePicture from '../../images/ProfilePicture.png';
import AmericanFlag from '../../images/AmericanFlag.png';
import { numberFormatter } from '../SocialReach';
import './style.css';

const countryArray = {
  us: AmericanFlag,
};

// Date format in JSON is depreciated and I couldn't quickly find any easy package
// that changed it for me so this ugly function makes it presentable.
const convertTimeToCorrectDate = (time) => {
  if (time === 'n/a') return 'n/a';
  if (!time) return '';
  const timeString = time.split(' ');
  const dateString = timeString[0].split('-');
  const hourString = Number(timeString[1].split(':')[0]);
  let realDay;
  if (hourString < 7) {
    if (dateString[1] < '10') {
      realDay = `0${(Number(dateString[1]) - 1)}`;
    } else {
      realDay = (Number(dateString[1]) - 1);
    }
  } else {
    realDay = dateString[1];
  }
  return moment(`${dateString[2]}-${dateString[0]}-${realDay}`).format('MMM D, YYYY');
};

class style extends PureComponent {
  render() {
    const { channelBaseInfo } = this.props;
    return (
      <div>
        <div className={'channel-base-info-container'}>
          <img alt="" src={ProfilePicture} />
          <div>
            <div className={'channel-name-container'}>
              <a href={`https://youtube.com/${channelBaseInfo.name}`}>
                {channelBaseInfo.name}&nbsp;&nbsp;&nbsp;
              </a>
              <div
                className={'linked-status'}
                style={{ backgroundColor: channelBaseInfo.status === 'linked' ? '#80d324' : 'red' }}
              >{channelBaseInfo.status}
              </div>
              <p>
                {channelBaseInfo.youtube_channel_id}
              </p>
            </div>
            <div className={'base-channel-youtube-stats-container'}>
              <div className={'views-subs-videos-container'}>
                <p><strong>{numberFormatter(channelBaseInfo.total_views)}</strong> Views</p>
                <p><strong>{numberFormatter(channelBaseInfo.subscribers)}</strong> Subs</p>
                <p><strong>{numberFormatter(channelBaseInfo.videos)}</strong> Videos</p>
              </div>
              <div className={'current-month-views-container'}>
                <p><strong>{numberFormatter(channelBaseInfo.last_month_views)}</strong><br />
                  &nbsp;{moment().subtract(1, 'months').format('MMM')} Views
                </p>
              </div>
            </div>
          </div>
        </div>
        <div className={'network-container'}>
          <div>
            <p><strong>Network:</strong> {channelBaseInfo.network}</p>
            <p><strong>Joined:</strong> {convertTimeToCorrectDate(channelBaseInfo.joined_at)}</p>
          </div>
          <div>
            <p><strong>Owner:</strong> {channelBaseInfo.owner}</p>
            <p><strong>Category:</strong> {channelBaseInfo.category}</p>
          </div>
          <div>
            <p><strong>Commission:</strong> {channelBaseInfo.commission}%</p>
            <p>
              <strong>Country: </strong>
              <img alt="" src={countryArray[channelBaseInfo.country_id]} />
            </p>
          </div>
        </div>
      </div>
    );
  }
}

style.propTypes = {
  channelBaseInfo: PropTypes.shape({
    category: PropTypes.string.isRequired,
    commission: PropTypes.number.isRequired,
    country_id: PropTypes.string.isRequired,
    joined_at: PropTypes.string.isRequired,
    last_month_views: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    network: PropTypes.string.isRequired,
    owner: PropTypes.string.isRequired,
    status: PropTypes.string.isRequired,
    subscribers: PropTypes.number.isRequired,
    total_views: PropTypes.number.isRequired,
    videos: PropTypes.number.isRequired,
    youtube_channel_id: PropTypes.string.isRequired,
  }).isRequired,
};


const mapStateToProps = ({ channelBaseInfo }) => ({
  channelBaseInfo,
});

const ChannelBaseInfo = connect(
  mapStateToProps,
  null,
)(style);

export default ChannelBaseInfo;
