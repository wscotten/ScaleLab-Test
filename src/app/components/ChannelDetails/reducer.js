import { GET_CHANNEL_INFO } from '../ChannelBaseInfo/constants';
import { PUT_SOCIAL_REACH_INFO_SUCCESS } from '../../containers/SocialReachModalButton/constants';
import { MONTHS_TO_SHOW } from '../ChannelDetails';

export const extractSocialReachNumbers = payload => ({
  total: payload.channelStats.social_reach,
  Youtube: payload.channelStats.youtube_reach,
  Facebook: payload.channelStats.facebook_reach,
  Vine: payload.channelStats.vine_reach,
  Twitter: payload.channelStats.twitter_reach,
  Instagram: payload.channelStats.instagram_reach,
});

const channelDetailsReducer = (data = {}, { type, payload }) => {
  switch (type) {
    case GET_CHANNEL_INFO:
      return {
        ageGroups: JSON.parse(payload.channelStats.age_groups),
        genders: {
          male: payload.channelStats.male_ratio,
          female: payload.channelStats.female_ratio,
        },
        socialReach: extractSocialReachNumbers(payload),
        earnings: payload.earnings.slice(payload.earnings.length - MONTHS_TO_SHOW),
      };
    case PUT_SOCIAL_REACH_INFO_SUCCESS:
      return Object.assign({}, data, { socialReach: payload });
    default:
      return data;
  }
};

export default channelDetailsReducer;
