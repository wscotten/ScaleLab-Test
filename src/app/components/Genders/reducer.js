import { GET_CHANNEL_INFO } from '../ChannelBaseInfo/reducer';

const gendersReducer = (genders = {}, { type, payload }) => {
  switch (type) {
    case GET_CHANNEL_INFO:
      return {
        male: payload.channelStats.male_ratio,
        female: payload.channelStats.female_ratio,
      };
    default:
      return genders;
  }
};

export default gendersReducer;
