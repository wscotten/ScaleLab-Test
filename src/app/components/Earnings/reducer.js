import { GET_CHANNEL_INFO } from '../ChannelBaseInfo/reducer';
import { MONTHS_TO_SHOW } from '../ChannelDetails';


const earningsReducer = (earnings = [], { type, payload }) => {
  switch (type) {
    case GET_CHANNEL_INFO:
      return payload.earnings.slice(payload.earnings.length - MONTHS_TO_SHOW);
    default:
      return earnings;
  }
};

export default earningsReducer;
