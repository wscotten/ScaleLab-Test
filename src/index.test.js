import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './app/index';
import store from './index';
import { getChannelInfo } from './app/components/ChannelBaseInfo/reducer';

const defaultData = {
  channels: {
    id: 78,
    youtube_channel_id: 'UCh8mkK_A3827d98sfslfa',
    name: 'RetroSnickers',
    network: 'ScaleLab',
    owner: 'Scalelab Affiliate',
    status: 'linked',
    country_id: 'us',
    joined_at: '03-24-2013 04:33:03',
    category: 'Entertainment',
    commission: 50.0,
  },
  channelStats: {
    id: 22,
    channel_id: 78,
    total_views: 9999982,
    last_month_views: 990148,
    subscribers: 7689,
    videos: 95,
    social_reach: 9900234,
    youtube_reach: 987654,
    facebook_reach: 99999,
    vine_reach: 50876,
    twitter_reach: 12987,
    instagram_reach: 1497,
    age_groups: '{"13": 9.8,"18": 30.1,"25": 29.8,"35": 10.1,"45": 8.0,"55": 4.9}',
    female_ratio: 59.8,
    male_ratio: 40.2,
  },
  earnings: [
    {
      id: 1,
      channel_id: 78,
      month: 3,
      year: 2016,
      gross: 12000,
    },
    {
      id: 2,
      channel_id: 79,
      month: 3,
      year: 2016,
      gross: 12000,
    },
    {
      id: 3,
      channel_id: 78,
      month: 4,
      year: 2016,
      gross: 15234,
    },
    {
      id: 4,
      channel_id: 78,
      month: 5,
      year: 2016,
      gross: 22213,
    },
    {
      id: 5,
      channel_id: 78,
      month: 6,
      year: 2016,
      gross: 12000,
    },
    {
      id: 6,
      channel_id: 78,
      month: 7,
      year: 2016,
      gross: 15234,
    },
    {
      id: 7,
      channel_id: 78,
      month: 8,
      year: 2016,
      gross: 22213,
    },
    {
      id: 8,
      channel_id: 78,
      month: 9,
      year: 2016,
      gross: 12000,
    },
    {
      id: 9,
      channel_id: 78,
      month: 10,
      year: 2016,
      gross: 15234,
    },
    {
      id: 10,
      channel_id: 78,
      month: 11,
      year: 2016,
      gross: 22213,
    },
    {
      id: 11,
      channel_id: 78,
      month: 12,
      year: 2016,
      gross: 49876,
    },
    {
      id: 12,
      channel_id: 78,
      month: 1,
      year: 2017,
      gross: 98723,
    },
    {
      id: 13,
      channel_id: 78,
      month: 2,
      year: 2017,
      gross: 10,
    },
  ],
};

store.dispatch(getChannelInfo(defaultData));

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>,
    div);
});

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
