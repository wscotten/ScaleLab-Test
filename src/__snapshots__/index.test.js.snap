// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="dashboard-container"
    >
      <div
        style={
          Object {
            "paddingLeft": 15,
            "paddingRight": 15,
          }
        }
      >
        <h4>
          Channels
        </h4>
      </div>
      <div>
        <div
          className="channel-base-info-container"
        >
          <img
            alt=""
            src="ProfilePicture.png"
          />
          <div>
            <div
              className="channel-name-container"
            >
              <a
                href="https://youtube.com/RetroSnickers"
              >
                RetroSnickers
                   
              </a>
              <div
                className="linked-status"
                style={
                  Object {
                    "backgroundColor": "#80d324",
                  }
                }
              >
                linked
              </div>
              <p>
                UCh8mkK_A3827d98sfslfa
              </p>
            </div>
            <div
              className="base-channel-youtube-stats-container"
            >
              <div
                className="views-subs-videos-container"
              >
                <p>
                  <strong>
                    10M
                  </strong>
                   Views
                </p>
                <p>
                  <strong>
                    7.7k
                  </strong>
                   Subs
                </p>
                <p>
                  <strong>
                    95
                  </strong>
                   Videos
                </p>
              </div>
              <div
                className="current-month-views-container"
              >
                <p>
                  <strong>
                    990k
                  </strong>
                  <br />
                   
                  Jul
                   Views
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="network-container"
        >
          <div>
            <p>
              <strong>
                Network:
              </strong>
               
              ScaleLab
            </p>
            <p>
              <strong>
                Joined:
              </strong>
               
              Mar 23, 2013
            </p>
          </div>
          <div>
            <p>
              <strong>
                Owner:
              </strong>
               
              Scalelab Affiliate
            </p>
            <p>
              <strong>
                Category:
              </strong>
               
              Entertainment
            </p>
          </div>
          <div>
            <p>
              <strong>
                Commission:
              </strong>
               
              50
              %
            </p>
            <p>
              <strong>
                Country: 
              </strong>
              <img
                alt=""
                src="AmericanFlag.png"
              />
            </p>
          </div>
        </div>
      </div>
      <div
        className="details-button-container"
      >
        <div
          className="detailsButton"
          onClick={[Function]}
          role="button"
          tabIndex={0}
        >
          Details 
          <svg>
            <path
              d="M0 0 L8 10 L16 0"
            />
          </svg>
        </div>
      </div>
      <div
        className="channel-stats-container"
      >
        <div
          className="social-reach-age-groups-genders-container"
        >
          <div
            className="social-reach-container"
          >
            <div
              className="social-reach-text"
            >
              <h4>
                Social Reach 
                9.9M
              </h4>
              <div
                className="social-reach-edit-button"
                onClick={[Function]}
                role="button"
                tabIndex={0}
              >
                <h4>
                  Edit
                </h4>
              </div>
            </div>
            <div
              className="social-reach-bar-container"
            >
              <div
                className="social-reach-bars"
              >
                <div>
                  <img
                    alt=""
                    src="YoutubeLogo.png"
                  />
                </div>
                <p>
                  988k
                </p>
              </div>
              <div
                className="social-reach-bars"
              >
                <div>
                  <img
                    alt=""
                    src="FacebookLogo.png"
                  />
                </div>
                <p>
                  100k
                </p>
              </div>
              <div
                className="social-reach-bars"
              >
                <div>
                  <img
                    alt=""
                    src="VineLogo.png"
                  />
                </div>
                <p>
                  51k
                </p>
              </div>
              <div
                className="social-reach-bars"
              >
                <div>
                  <img
                    alt=""
                    src="TwitterLogo.png"
                  />
                </div>
                <p>
                  13k
                </p>
              </div>
              <div
                className="social-reach-bars"
              >
                <div>
                  <img
                    alt=""
                    src="InstagramLogo.png"
                  />
                </div>
                <p>
                  1.5k
                </p>
              </div>
            </div>
          </div>
          <div
            className="age-group-container"
          >
            <h4>
              Age Groups
            </h4>
            <div
              className="age-group-bar-container"
            >
              <div>
                <p
                  style={
                    Object {
                      "color": "#f8e71c",
                    }
                  }
                >
                  10%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#f8e71c",
                      "height": 26.53333333333333,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  13-17
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#b9e986",
                    }
                  }
                >
                  30%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#b9e986",
                      "height": 40,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  18-24
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#7ed321",
                    }
                  }
                >
                  30%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#7ed321",
                      "height": 39.86666666666667,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  25-34
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#50e3c1",
                    }
                  }
                >
                  10%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#50e3c1",
                      "height": 26.733333333333334,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  35-34
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#1abbcf",
                    }
                  }
                >
                  8%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#1abbcf",
                      "height": 25.333333333333332,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  45-54
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#4a91e3",
                    }
                  }
                >
                  5%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#4a91e3",
                      "height": 23.266666666666666,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  55+
                </p>
              </div>
            </div>
          </div>
          <div
            className="genders-container"
          >
            <h4>
              Genders
            </h4>
            <div
              className="genders-bar-container"
            >
              <div>
                <p
                  style={
                    Object {
                      "color": "#1293f4",
                    }
                  }
                >
                  40%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#1293f4",
                      "height": 30.1,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  male
                </p>
              </div>
              <div>
                <p
                  style={
                    Object {
                      "color": "#bd0de0",
                    }
                  }
                >
                  60%
                </p>
                <div
                  style={
                    Object {
                      "backgroundColor": "#bd0de0",
                      "height": 39.9,
                      "margin": "auto",
                      "width": 30,
                    }
                  }
                />
                <p>
                  female
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="earnings-container"
        >
          <h4>
            Earnings
          </h4>
          <div
            className="earnings-bar-container"
          >
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $12,000
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Jun
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $15,234
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Jul
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $22,213
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Aug
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $12,000
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Sep
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $15,234
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Oct
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $22,213
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Nov
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $49,876
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 45.156346545384565,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Dec
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $98,723
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 60,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Jan
              </p>
            </div>
            <div>
              <p
                style={
                  Object {
                    "color": "#1293f4",
                  }
                }
              >
                $10
              </p>
              <div
                style={
                  Object {
                    "backgroundColor": "#1293f4",
                    "height": 30,
                    "margin": "auto",
                    "width": 30,
                  }
                }
              />
              <p>
                Feb
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
